File: web/src/api/internalSystem/futureData/departmentAccountProduct.js
import service from '@/utils/request'

// @Tags DepartmentAccountProduct
// @Summary 创建DepartmentAccountProduct
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DepartmentAccountProduct true "创建DepartmentAccountProduct"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /departmentAccountProduct/createDepartmentAccountProduct [post]
export const createDepartmentAccountProduct = (data) => {
     return service({
         url: "/departmentAccountProduct/createDepartmentAccountProduct",
         method: 'post',
         data
     })
 }


// @Tags DepartmentAccountProduct
// @Summary 删除DepartmentAccountProduct
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DepartmentAccountProduct true "删除DepartmentAccountProduct"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /departmentAccountProduct/deleteDepartmentAccountProduct [delete]
 export const deleteDepartmentAccountProduct = (data) => {
     return service({
         url: "/departmentAccountProduct/deleteDepartmentAccountProduct",
         method: 'delete',
         data
     })
 }

// @Tags DepartmentAccountProduct
// @Summary 删除DepartmentAccountProduct
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除DepartmentAccountProduct"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /departmentAccountProduct/deleteDepartmentAccountProduct [delete]
 export const deleteDepartmentAccountProductByIds = (data) => {
     return service({
         url: "/departmentAccountProduct/deleteDepartmentAccountProductByIds",
         method: 'delete',
         data
     })
 }

// @Tags DepartmentAccountProduct
// @Summary 更新DepartmentAccountProduct
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DepartmentAccountProduct true "更新DepartmentAccountProduct"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /departmentAccountProduct/updateDepartmentAccountProduct [put]
 export const updateDepartmentAccountProduct = (data) => {
     return service({
         url: "/departmentAccountProduct/updateDepartmentAccountProduct",
         method: 'put',
         data
     })
 }


// @Tags DepartmentAccountProduct
// @Summary 用id查询DepartmentAccountProduct
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DepartmentAccountProduct true "用id查询DepartmentAccountProduct"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /departmentAccountProduct/findDepartmentAccountProduct [get]
 export const findDepartmentAccountProduct = (params) => {
     return service({
         url: "/departmentAccountProduct/findDepartmentAccountProduct",
         method: 'get',
         params
     })
 }


// @Tags DepartmentAccountProduct
// @Summary 分页获取DepartmentAccountProduct列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取DepartmentAccountProduct列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /departmentAccountProduct/getDepartmentAccountProductList [get]
 export const getDepartmentAccountProductList = (params) => {
     return service({
         url: "/departmentAccountProduct/getDepartmentAccountProductList",
         method: 'get',
         params
     })
 }

File: server/router/internal_system/future_data/departmentAccountProduct.go
package future_data

import (
	v1 "gin-vue-admin/api/v1/internal_system/future_data"
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

func InitDepartmentAccountProductRouter(Router *gin.RouterGroup) {
	DepartmentAccountProductRouter := Router.Group("departmentAccountProduct").Use(middleware.OperationRecord())
	{
		DepartmentAccountProductRouter.POST("createDepartmentAccountProduct", v1.CreateDepartmentAccountProduct)             // 新建DepartmentAccountProduct
		DepartmentAccountProductRouter.DELETE("deleteDepartmentAccountProduct", v1.DeleteDepartmentAccountProduct)           // 删除DepartmentAccountProduct
		DepartmentAccountProductRouter.DELETE("deleteDepartmentAccountProductByIds", v1.DeleteDepartmentAccountProductByIds) // 批量删除DepartmentAccountProduct
		DepartmentAccountProductRouter.PUT("updateDepartmentAccountProduct", v1.UpdateDepartmentAccountProduct)              // 更新DepartmentAccountProduct
		DepartmentAccountProductRouter.GET("findDepartmentAccountProduct", v1.FindDepartmentAccountProduct)                  // 根据ID获取DepartmentAccountProduct
		DepartmentAccountProductRouter.GET("getDepartmentAccountProductList", v1.GetDepartmentAccountProductList)            // 获取DepartmentAccountProduct列表
	}
}
